lib/ansible/modules/network/fortios/fortios_system_resource_limits.py

```diff
@@ -0,0 +1,395 @@
+        for k, v in data.items():
+            new_data[k.replace('_', '-')] = underscore_to_hyphen(v)
+        data = new_data
+
+    return data
+
+
+def system_resource_limits(data, fos):
+    vdom = data['vdom']
+    system_resource_limits_data = data['system_resource_limits']
+    filtered_data = underscore_to_hyphen(filter_system_resource_limits_data(system_resource_limits_data))
+
+    return fos.set('system',
+                   'resource-limits',
+                   data=filtered_data,
+                   vdom=vdom)
+
+
+def is_successful_status(status):
+    return status['status'] == "success" or \
+        status['http_method'] == "DELETE" and status['http_status'] == 404
+
+
+def fortios_system(data, fos):
+
+    if data['system_resource_limits']:
+        resp = system_resource_limits(data, fos)
+
+    return not is_successful_status(resp), \
+        resp['status'] == "success", \
+        resp
+
+
+def main():
+    fields = {
+        "host": {"required": False, "type": "str"},
+        "username": {"required": False, "type": "str"},
+        "password": {"required": False, "type": "str", "no_log": True},
+        "vdom": {"required": False, "type": "str", "default": "root"},
+        "https": {"required": False, "type": "bool", "default": True},
+        "ssl_verify": {"required": False, "type": "bool", "default": True},
+        "system_resource_limits": {
+            "required": False, "type": "dict", "default": None,
+            "options": {
+                "custom_service": {"required": False, "type": "int"},
+                "dialup_tunnel": {"required": False, "type": "int"},
+                "firewall_address": {"required": False, "type": "int"},
+                "firewall_addrgrp": {"required": False, "type": "int"},
+                "firewall_policy": {"required": False, "type": "int"},
+                "ipsec_phase1": {"required": False, "type": "int"},
+                "ipsec_phase1_interface": {"required": False, "type": "int"},
+                "ipsec_phase2": {"required": False, "type": "int"},
+                "ipsec_phase2_interface": {"required": False, "type": "int"},
+                "log_disk_quota": {"required": False, "type": "int"},
+                "onetime_schedule": {"required": False, "type": "int"},
+                "proxy": {"required": False, "type": "int"},
+                "recurring_schedule": {"required": False, "type": "int"},
+                "service_group": {"required": False, "type": "int"},
+                "session": {"required": False, "type": "int"},
+                "sslvpn": {"required": False, "type": "int"},
+                "user": {"required": False, "type": "int"},
+                "user_group": {"required": False, "type": "int"}
+
+            }
+        }
+    }
+
+    module = AnsibleModule(argument_spec=fields,
+                           supports_check_mode=False)
+
+    legacy_mode = 'host' in module.params and module.params['host'] is not None and \
```

Comment:
it might be worth adding a note about this legacy mode. Which version does it affect?

Reply:
This refers to using fortiosapi instead of HTTPAPI. I can add a comment stating that if you agree.

Reply:
works for me! thanks!